GVRf/Framework/jni/objects/components/texture_capturer.cpp

```diff
@@ -164,8 +167,10 @@ glm::mat4 TextureCapturer::getMvpMatrix(float half_width, float half_height) {
 }
 
 void TextureCapturer::callback(int msg, char *info) {
-    Java_org_gearvrf_NativeTextureCapturer_callbackFromNative(
-            mJNIEnv, mCapturerObject, msg, info);
+    if (nullptr != mCapturerObject) {
```

Comment:
if mCapturerObject is r/w, a mutex is needed to protect it here. race condition can occur between this if-statement and Lines 48 - 55.